jacodb-analysis/src/main/kotlin/org/jacodb/analysis/codegen/language/impl/JavaLanguage.kt

```diff
@@ -421,23 +433,51 @@ class JavaLanguage : TargetLanguage {
 
     private fun appendStartFunction(name: String, function: FunctionPresentation) {
         writeSeparated("public class ${classNameForStaticFunction(name)}")
-        inScope {
+        inScope({
             tabulate()
             writeVisibility(function.visibility)
             writeSeparated("static")
             writeTypeUsage(function.returnType)
             write(function.shortName)
             writeParametersList(function)
-            inScope {
+            inScope({
                 appendLocalsAndSites(function)
+            }, true)
+            if (classNameForStaticFunction(name).contains("ClassForStartFunctionForNpeInstance")) {
+                write(SEPARATOR.repeat(2))
+                tabulate()
+                writeVisibility(VisibilityModifier.PUBLIC)
+                writeSeparated("static")
+                writeSeparated("void")
+                writeSeparated("main(String[] args)")
```

Comment:
this is not how psvm was intendent to be added. You should have added it as tree structure using TypeRepresentation.staticCounterPart